src/Symfony/Component/Notifier/Bridge/OneSignal/OneSignalTransport.php

```diff
@@ -0,0 +1,105 @@
+
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\PushMessage;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Tomas Norkūnas <norkunas.tom@gmail.com>
+ */
+final class OneSignalTransport extends AbstractTransport
+{
+    protected const HOST = 'onesignal.com';
+
+    private $appId;
+    private $apiKey;
+
+    public function __construct(string $appId, string $apiKey, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->appId = $appId;
+        $this->apiKey = $apiKey;
+        $this->client = $client;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('onesignal://%s?app_id=%s', $this->getEndpoint(), urlencode($this->appId));
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof PushMessage && (null === $message->getOptions() || $message->getOptions() instanceof OneSignalOptions);
+    }
+
+    /**
+     * @see https://documentation.onesignal.com/reference/create-notification
+     */
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof PushMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, PushMessage::class, $message);
+        }
+
+        if ($message->getOptions() && !$message->getOptions() instanceof OneSignalOptions) {
+            throw new LogicException(sprintf('The "%s" transport only supports instances of "%s" for options.', __CLASS__, OneSignalOptions::class));
+        }
+
+        if (!($opts = $message->getOptions()) && $notification = $message->getNotification()) {
+            $opts = OneSignalOptions::fromNotification($notification);
+        }
+
+        $options = $opts ? $opts->toArray() : [];
+        $options['app_id'] = $this->appId;
+
+        if (!isset($options['headings'])) {
+            $options['headings'] = ['en' => $message->getSubject()];
+        }
+        if (!isset($options['contents'])) {
+            $options['contents'] = ['en' => $message->getContent()];
+        }
+
+        $options['include_player_ids'] = [$message->getRecipientId()];
+
+        $response = $this->client->request('POST', 'https://'.$this->getEndpoint().'/api/v1/notifications', [
+            'headers' => [
+                'Accept' => 'application/json',
+                'Authorization' => 'Basic '.$this->apiKey,
+            ],
+            'json' => array_filter($options),
```

Comment:
```suggestion
            'json' => array_filter($options),
```

array_filter really needed?